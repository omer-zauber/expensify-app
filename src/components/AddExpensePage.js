import React from 'react';
import ReactDOM from 'react-dom';

const AddExpensePage = () => (
  <div>
    This is the add exchange component.
  </div>
);

export default AddExpensePage;