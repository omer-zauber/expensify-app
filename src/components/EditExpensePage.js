import React from 'react';
import ReactDOM from 'react-dom';

const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      This is the edit component. Editing expense id {props.match.params.id}.
    </div>   
  );
}

export default EditExpensePage;