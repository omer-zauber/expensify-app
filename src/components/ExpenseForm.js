import React from 'react';

export default class ExpenseForm extends React.Component {
  state = {
    description: ''
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(()=> ({ description }));
  };

  render() {
    return (
      <div>
        <form action="">
          <input 
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="number"
            placeholder="Amount"
          />
          <textarea name="" id="" cols="30" rows="10" placeholder="Add a note for your expense (optional)"></textarea>
          <button type="submit">Add Expense</button>
        </form>
      </div>
    )
  };
};