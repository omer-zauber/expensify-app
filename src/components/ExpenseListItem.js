import React from 'react';
import { Link } from 'react-router-dom'
import moment from 'moment';
import numeral from 'numeral';

numeral.register('locale', 'il', {
  delimiters: {
    thousands: ',',
    decimal: '.'
  },
  currency: {
    symbol: 'â‚ª'
  }
});

numeral.locale('il');

const ExpenseListItem = ({  id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount/100).format('$0,0.00')}
      -
      {moment(createdAt).format('D MMMM, YYYY')}
    </p>        
  </div>
);

export default ExpenseListItem;