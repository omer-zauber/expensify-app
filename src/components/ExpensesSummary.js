import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export class ExpensesSummary extends React.Component {
  state = {

    total: selectExpensesTotal(this.props.expenses)
  };

  render() {
    return (
      <div>
        {
          this.props.expenses.length > 0 &&
          <p>
            Viewing {this.props.expenses.length} expense{this.props.expenses.length > 1 ? 's' : ''} totalling {numeral(selectExpensesTotal(this.props.expenses) / 100).format('$0,0.00')}
          </p>
        }
      </div>
    );
  };
}


const mapStateToProps = (state) => ({expenses: selectExpenses(state.expenses, state.filters)});

export default connect(mapStateToProps)(ExpensesSummary);