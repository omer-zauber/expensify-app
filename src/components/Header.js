import React from 'react';
import ReactDOM from 'react-dom';
import { Link, NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <p>
      <NavLink to ="/" activeClassName="is-active" exact={true}>Home</NavLink> |
      <NavLink to ="/create" activeClassName="is-active" exact={true}>Create</NavLink> |
      <NavLink to ="/edit" activeClassName="is-active" exact={true}>Edit</NavLink> |
      <NavLink to ="/help" activeClassName="is-active" exact={true}>Help</NavLink>     
    </p>
  </header>
);

export default Header;