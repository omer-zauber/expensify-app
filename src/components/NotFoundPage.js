import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from "react-router-dom";

const NotFoundPage = () => (
  <div>
    This is the 404 component.
    <Link to ="/">Go Home.</Link>
  </div>
);

export default NotFoundPage;