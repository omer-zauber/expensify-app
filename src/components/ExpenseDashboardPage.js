import React from 'react';
import ExpensesSummary from './ExpensesSummary';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpeneseListFilters'


const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters /> 
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;