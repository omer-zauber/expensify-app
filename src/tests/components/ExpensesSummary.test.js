import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test('should render correct message for 0 expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correct message for one expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[2]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correct message for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});