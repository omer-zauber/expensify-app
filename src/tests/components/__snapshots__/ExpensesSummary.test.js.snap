// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct message for 0 expenses 1`] = `<div />`;

exports[`should render correct message for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render correct message for one expenses 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $45.00
  </p>
</div>
`;
